proxy/src/control/destination/background.rs

```diff
@@ -316,6 +326,15 @@ where
         }
     }
 
+    fn poll_tls_config_updates(&mut self) {
+        if let Ok(Async::Ready(_)) = self.client_tls.poll() {
+            self.tls_config_version += 1;
+            for (auth, set) in &mut self.destinations {
+                set.set_tls_config_version(auth, self.tls_config_version);
```

Comment:
Instead of setting the TLS configuration version on each `DestinationSet` here, so that later the DestinationSet will report that every entry changed, can't we just immediately report that every (TLS) entry in every DestinationSet changed, without actually modifying the contents of any DestinationSet?

Reply:
If I understand correctly, this would require that we invalidate endpoints that do not need to be rebound for TLS (i.e. if no TLS is configured for the endpoint).

Reply:
I think we could just have the destination set loop over every endpoint and send `Modification` events for only the TLS ones.

Reply:
Right, we'd only do so for TLS endpoints, not non-TLS endpoints. (It isn't urgent to skip rebinding the non-TLS endpoints anyway, as that's purely an optimization, we don't expect many non-TLS connections, and this event is rare.)